File: backend/src/main/java/com/quality/platform/util/JsonUtils.java
package com.quality.platform.util;

import com.fasterxml.jackson.core.JsonProcessingException;
import com.fasterxml.jackson.core.type.TypeReference;
import com.fasterxml.jackson.databind.ObjectMapper;
import lombok.extern.slf4j.Slf4j;

import java.util.Map;

@Slf4j
public class JsonUtils {
    private static final ObjectMapper mapper = new ObjectMapper();

    public static String toJson(Object obj) {
        try {
            return mapper.writeValueAsString(obj);
        } catch (JsonProcessingException e) {
            log.error("toJson error", e);
            return null;
        }
    }

    public static <T> T fromJson(String json, Class<T> clazz) {
        try {
            return mapper.readValue(json, clazz);
        } catch (JsonProcessingException e) {
            log.error("fromJson error", e);
            return null;
        }
    }

    public static Map<String, Object> parseMap(String json) {
        try {
            return mapper.readValue(json, new TypeReference<Map<String, Object>>() {});
        } catch (JsonProcessingException e) {
            log.error("parseMap error", e);
            return null;
        }
    }
}

File: spark-streaming-quality-job/src/main/scala/com/quality/streaming/utils/JsonUtils.scala
package com.quality.streaming.utils

import com.fasterxml.jackson.core.`type`.TypeReference
import com.fasterxml.jackson.databind.{DeserializationFeature, ObjectMapper}
import com.fasterxml.jackson.module.scala.DefaultScalaModule
import com.fasterxml.jackson.module.scala.experimental.ScalaObjectMapper

import scala.collection.immutable.{Map => ImmutableMap}

object JsonUtils {
  private val mapper = new ObjectMapper() with ScalaObjectMapper
  mapper.registerModule(DefaultScalaModule)
  mapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)

  def toJson(obj: Any): String = {
    mapper.writeValueAsString(obj)
  }

  def parseMap(json: String): ImmutableMap[String, Any] = {
    mapper.readValue[ImmutableMap[String, Any]](json)
  }

  def parseList(json: String): List[ImmutableMap[String, Any]] = {
    mapper.readValue(json, new TypeReference[List[ImmutableMap[String, Any]]] {})
  }

  def parse[T](json: String, clazz: Class[T]): T = {
    mapper.readValue(json, clazz)
  }
}
